stubs/oauthlib/oauthlib/oauth2/rfc6749/request_validator.pyi

```diff
@@ -1,8 +1,21 @@
-from typing import Any
+from typing import Any, Literal, TypedDict
+from typing_extensions import NotRequired
 
 from oauthlib.common import Request
 from oauthlib.oauth2.rfc6749.clients import Client
 
+class _BearerToken(TypedDict):
+    token_type: Literal["Bearer"]
+    access_token: str
+    expires_in: int
+    scope: str
+    refresh_token: NotRequired[str]
+    state: NotRequired[str]
+
+class _AuthorizationCode(TypedDict):
+    code: str
+    state: NotRequired[str]
```

Comment:
The docs suggest there is also an optional `nonce` key: https://oauthlib.readthedocs.io/en/latest/oauth2/validator.html?highlight=save_authorization_code#oauthlib.oauth2.RequestValidator.save_authorization_code

Reply:
Good catch